src/routes/transactions/mappers/multisig-transactions/multisig-transaction-details.mapper.ts

```diff
@@ -0,0 +1,230 @@
+        transaction.operation,
+        isTrustedDelegateCall,
+        isEmpty(addressInfoIndex) ? null : addressInfoIndex,
+      ),
+      txHash: transaction.transactionHash,
+      detailedExecutionInfo,
+      safeAppInfo,
+    };
+  }
+
+  private async _mapMultisigExecutionDetails(
+    chainId: string,
+    transaction: MultisigTransaction,
+    txStatus: TransactionStatus,
+    safe: Safe,
+  ): Promise<MultisigExecutionDetails> {
+    const gasToken = transaction.gasToken ?? NULL_ADDRESS;
+    const gasTokenInfo =
+      gasToken != NULL_ADDRESS
+        ? await this.tokenRepository.getToken(chainId, gasToken)
+        : null;
+    const executor = transaction.executor
+      ? await this.addressInfoHelper.getOrDefault(chainId, transaction.executor)
+      : null;
+    const signers = safe.owners.map((owner) => new AddressInfo(owner));
+    const confirmationsRequired =
+      transaction.confirmationsRequired ?? safe.threshold;
+    const confirmations = !transaction.confirmations
+      ? []
+      : transaction.confirmations.map(
+          (confirmation) =>
+            new MultisigConfirmationDetails(
+              new AddressInfo(confirmation.owner),
+              confirmation.signature,
+              confirmation.submissionDate.getTime(),
+            ),
+        );
+    const refundReceiver = await this.addressInfoHelper.getOrDefault(
+      chainId,
+      transaction.refundReceiver ?? NULL_ADDRESS,
+    );
+    const rejectors =
+      txStatus === TransactionStatus.Cancelled
+        ? await this._getRejectors(chainId, transaction, safe)
+        : null;
+
+    return {
+      submittedAt: transaction.submissionDate.getTime(),
+      nonce: transaction.nonce,
+      safeTxGas: transaction.safeTxGas?.toString() ?? '0',
+      baseGas: transaction.baseGas?.toString() ?? '0',
+      gasPrice: transaction.gasPrice?.toString() ?? '0',
+      refundReceiver,
+      safeTxHash: transaction.safeTxHash,
+      executor,
+      signers,
+      confirmationsRequired,
+      confirmations,
+      rejectors,
+      gasToken,
+      gasTokenInfo,
+      trusted: transaction.trusted,
+    };
+  }
+
+  /**
+   * Tries to get the {@link AddressInfo} related to the address from a Token, and
+   * fallbacks to a Contract. If none can be found, a default {@link AddressInfo}
+   * containing just the input address is returned.
+   *
+   * @param chainId - chain id to use
+   * @param address - the address of the source to which we want to retrieve its metadata
+   * @returns {@link AddressInfo} containing the address metadata
+   */
+  private async _getRecipientAddressInfo(
+    chainId: string,
+    address: string,
+  ): Promise<AddressInfo> {
+    try {
+      const tokenAddressInfo = await this.addressInfoHelper.get(
+        chainId,
+        address,
+        'TOKEN',
+      );
+      return tokenAddressInfo ?? new AddressInfo(address);
+    } catch (err) {
+      return await this.addressInfoHelper.getOrDefault(
```

Comment:
Nit: you need not `await` this.

```suggestion
      return this.addressInfoHelper.getOrDefault(
```

Reply:
Changed in